build/commands/lib/pylint.js

```diff
@@ -0,0 +1,89 @@
+// Copyright (c) 2021 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// you can obtain one at http://mozilla.org/MPL/2.0/.
+
+const path = require('path')
+const fs = require('fs')
+const config = require('../lib/config')
+const util = require('../lib/util')
+
+const getDefaultOptions = () => {
+  let options = Object.assign({}, config.defaultOptions)
+  options.cwd = path.resolve(__dirname, '../../..')
+  return options
+}
+
+const runPylint = (args, continueOnFail = false) => {
+  let cmd_options = getDefaultOptions()
+  cmd_options.continueOnFail = continueOnFail
+  if (process.platform !== 'win32') {
+    util.run('which', ['pylint'], cmd_options)
+  } else {
+    util.run('where', ['pylint'], cmd_options)
+  }
+  util.run('pylint', ['--version'], cmd_options)
+  util.run('pylint', args, cmd_options)
+}
+
+const runGit = (gitArgs) => {
+  // Have to use shell because otherwise git gets confused with node's escaping
+  // of the args.
+  let prog = util.run('git', gitArgs, { shell: true })
+  const output = prog.stdout.toString()
+  console.log(output)
+  return output.trim()
+}
+
+const getChangedFiles = (base_branch, check_folders) => {
+  const merge_base = runGit(['merge-base', 'origin/' + base_branch, 'HEAD'])
+  if (!merge_base) {
+    console.log('Could not determine merge-base.')
+    process.exit(1)
+  }
+  const changed_files = runGit(['diff', '--name-only', '--diff-filter',
+    'drt', merge_base, '--', check_folders.join('/*.py ') + '/*.py'])
+  if (!changed_files) {
+    return []
+  }
+  return changed_files.split('\n')
+}
+
+const pylint = (options = {}) => {
+  const check_folders = ['build', 'components', 'installer', 'script', 'tools']
+  const report_file = 'pylint-report.txt'
+
+  let paths = check_folders
+  let description = 'pylint findings'
+  if (options.base) {
+    paths = getChangedFiles(options.base, check_folders)
+    description += ' in scripts changed relative to ' + options.base
+  }
+  description += ' in: ' + check_folders.join('/ ') + '/'
+
+  if (!paths.length) {
+    console.log('No ' + description)
+    if (options.report) {
+      try {
+        fs.writeFileSync(report_file, '\n')
+      } catch(err) {
+        console.log('Unable to write to report file: ' + err)
```

Comment:
nit: use `console.error`. Also err is likely to be an error object not a string, right? In which case provide as another argument to console.log instead of string concat as it may provide more info.

Reply:
Addressed in https://github.com/brave/brave-core/pull/8782/files#diff-51c6e3f7ee7e15fa6a0611b0c64b87a7ca9a800564bf30e0a9107326d3ef6039R83 and other places.